src/find/matchers/time.rs

```diff
@@ -0,0 +1,325 @@
+use std;
+use std::error::Error;
+use std::fs::{File, Metadata};
+use std::io::{stderr, Write};
+use std::time::SystemTime;
+use walkdir::DirEntry;
+
+use find::matchers::{ComparableValue, Matcher, MatcherIO};
+
+const SECS_PER_DAY: i64 = 60 * 60 * 24;
+
+/// This matcher checks whether a file is newer than the file the matcher is initialized with.
+pub struct NewerMatcher {
+    given_modification_time: SystemTime,
+}
+
+impl NewerMatcher {
+    pub fn new(path_to_file: &str) -> Result<NewerMatcher, Box<Error>> {
+        let f = try!(File::open(path_to_file));
+        let metadata = try!(f.metadata());
+        Ok(NewerMatcher { given_modification_time: try!(metadata.modified()) })
+    }
+
+    pub fn new_box(path_to_file: &str) -> Result<Box<Matcher>, Box<Error>> {
+        Ok(Box::new(try!(NewerMatcher::new(path_to_file))))
+    }
+
+    /// Impementation of matches that returns a result, allowing use to use try!
+    /// to deal with the errors.
+    fn matches_impl(&self, file_info: &DirEntry) -> Result<bool, Box<Error>> {
+        let this_time = try!(try!(file_info.metadata()).modified());
+        // duration_since returns an Ok duration if this_time <= given_modification_time
+        // and returns an Err (with a duration) otherwise. So if this_time >
+        // given_modification_time (in which case we want to return true) then
+        // duration_since will return an error.
+        Ok(self.given_modification_time.duration_since(this_time).is_err())
+    }
+}
+
+impl Matcher for NewerMatcher {
+    fn matches(&self, file_info: &DirEntry, _: &mut MatcherIO) -> bool {
+        match self.matches_impl(file_info) {
+            Err(e) => {
+                writeln!(&mut stderr(),
```

Comment:
Consider a `print_err!` macro for error printing; among other things, this would make it easier to abstract the actual error logging.